src/types/UiaTextRangeBase.cpp

```diff
@@ -1112,20 +1111,20 @@ IFACEMETHODIMP UiaTextRangeBase::MoveEndpointByUnit(_In_ TextPatternRangeEndpoin
     // If so, clamp each endpoint to the end of the document.
     const auto bufferSize{ _pData->GetTextBuffer().GetSize() };
 
-    auto documentEnd = bufferSize.EndExclusive();
+    til::point documentEnd{ bufferSize.EndExclusive().X, bufferSize.EndExclusive().Y };
     try
     {
-        documentEnd = _getDocumentEnd().to_win32_coord();
+        documentEnd = _getDocumentEnd();
     }
     CATCH_LOG();
 
-    if (bufferSize.CompareInBounds(_start, documentEnd, true) > 0)
+    if (til::point{ _start } > documentEnd)
     {
-        _start = documentEnd;
+        _start = documentEnd.to_win32_coord();
     }
-    if (bufferSize.CompareInBounds(_end, documentEnd, true) > 0)
+    if (til::point{ _end } > documentEnd)
     {
-        _end = documentEnd;
+        _end = documentEnd.to_win32_coord();

```

Comment:
i have to admit, I somewhat thought that "porting the coordinate system change for UIA only" would mean that `_start` and `_end` were natively `til::point`, and didn't have to be converted every time we encountered them

Reply:
I wish. But the problem is Viewport and TextBuffer still need them to be COORDs. And occasionally, they'll be copied by reference. It's doable, but I figured this would be the easiest path forward with the least risk. Especially since this is a servicing update after all haha